doc/concepts/configuration.rst

```diff
@@ -0,0 +1,651 @@
+
+    In this example, ``iproto.listen`` is in effect for all instances in *group001*.
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/group_scope/config.yaml
+        :language: yaml
+        :emphasize-lines: 3-4
+        :dedent:
+
+-   *Global*
+
+    In this example, ``iproto.listen`` is applied to all instances of all groups.
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/global_scope/config.yaml
+        :language: yaml
+        :emphasize-lines: 1-2
+        :dedent:
+
+
+.. NOTE::
+
+    The :ref:`Configuration reference <configuration_reference>` contains information to which scopes each configuration option can be applied.
+
+
+.. _configuration_replica_set_scopes:
+
+Configuration scopes: Replica set example
+*****************************************
+
+The example below shows how specific configuration options work in different configuration scopes for a replica set with a manual failover.
+You can learn more about configuring replication from :ref:`Replication tutorials <how-to-replication>`.
+
+..  literalinclude:: /code_snippets/snippets/replication/instances.enabled/manual_leader/config.yaml
+    :language: yaml
+    :emphasize-lines: 1-8,10-12,14-15,21,24-25,27-28,30-31
+    :end-before: Load sample data
+    :dedent:
+
+-   ``credentials`` (*global*)
+
+    Options in this section grant the specified roles to the *replicator* and *client* users.
+    These options are applied globally to all instances.
+
+-   ``iproto`` (*global*, *instance*)
+
+    The ``iproto`` section is specified on both global and instance levels.
+    The ``iproto.advertise.peer`` option specifies a URI used by an instance to connect to another instance as a replica.
+    In the example above, the URI includes a user name only.
+    A host value is taken from ``iproto.listen`` that is set on the instance level.
+
+-   ``replication``: (*global*)
+
+    The ``replication.failover`` global option sets a manual failover for all replica sets.
+
+-   ``leader``: (*replica set*)
+
+    The ``<replicaset-name>.leader`` option sets a :ref:`master <replication-roles>` instance for *replicaset001*.
+
+
+
+.. _configuration_application:
+
+Loading an application
+**********************
+
+Using Tarantool as an application server, you can write your own applications in Lua.
+In the ``app`` section, you can load the application and provide a custom application configuration in the ``cfg`` section.
+
+In the example below, the application is loaded from the ``myapp.lua`` file placed next to the YAML configuration file:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application/config.yaml
+    :language: yaml
+    :dedent:
+
+To get a value of the custom ``greeting`` property in the application code,
+use the ``config:get()`` function provided by the :ref:`config <config-module>` module.
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application/myapp.lua
```

Comment:
I'd start the code from a comment line with file name `.. myapp.lua` so that users can quickly see that it's the app file without reading the entire text.